publish-mqtt/src/main.rs

```diff
@@ -363,20 +424,13 @@ async fn connect_start_sensor<'a>(
         .await
         .with_context(|| std::line!().to_string())?;
```

Comment:
Should the `connection_status` be updated if this fails, rather than leaving it in the `Connecting` state?

Reply:
I think this is now unnecessary because the state will revert to what it was before it was updated to `Connecting`. Now I think the only reason the `Connecting` state has an effect is if an update for the sensor comes in before this function returns. We can probably solve this better with locking.